packages/asset-value/src/assetValue.ts

```diff
@@ -0,0 +1,212 @@
+import { AssetId } from '@shapeshiftoss/caip'
+import { BigNumber } from 'bignumber.js'
+import { Md5 } from 'ts-md5'
+
+import { CHECKSUM_LENGTH, DEFAULT_FORMAT_DECIMALS } from './constants'
+import { AssetValueParams, isAssetValueParams, SerializedAssetValue } from './types'
+import { AssetValueFormat } from './types'
+import { bn, bnOrZero } from './utils'
+
+type AssetValueState = {
+  assetId: AssetId
+  precision: number
+  value: string
+}
+
+//TODO(pastaghost): Add JSDoc documentation
+export class AssetValue {
```

Comment:
my only other thought is around the use of `bnOrZero` throughout in relation to being a big number wrapped asset. kind of makes it unpure in a sense, though it is for our use case and we use that for all of our bignumber math (I still have mixed feeling about this in general but it is what it is haha)